framework/classes/controller/front.ctrl.php

```diff
@@ -249,32 +250,46 @@ public function router($action, $params, $status = 200)
                     exit($e->getMessage());
                 }
             }
-
-            if ($_404) {
-                \Event::trigger('front.404NotFound', array('url' => rtrim($this->_page_url, '/')));
-
-                // If no redirection then we display 404
-                if (!empty($url)) {
-                    $_SERVER['NOS_URL'] = '';
-
-                    return $this->router('index', $params, 404);
-                } else {
-                    // The DB config is there, there's probably no homepage.
-                    echo \View::forge('nos::errors/blank_slate_front', array(
-                        'base_url' => $this->_base_href,
-                    ), false);
-                    exit();
-                }
-            }
         }
 
+        // Disable browser cache if it's a preview
         if ($this->_is_preview) {
             $this->setHeader('Cache-Control', 'no-cache, no-store, must-revalidate');
             $this->setHeader('Pragma', 'no-cache');
             $this->setHeader('Expires', '0');
```

Comment:
Header Expires *must* be a valid date (eg. Sat, 26 Jul 1997 05:00:00 GMT), see http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.21  
Max-Age can also be set to 0 as an additional measure against caching proxies